backend/flow_api/flow/registration/action_register_login_identifier.go

```diff
@@ -118,15 +118,13 @@ func (a RegisterLoginIdentifier) Execute(c flowpilot.ExecutionContext) error {
 		if deps.Cfg.Password.Enabled {
 			return c.StartSubFlow(passcode.StatePasscodeConfirmation, StatePasswordCreation)
 		} else if !deps.Cfg.Passcode.Enabled || (deps.Cfg.Passkey.Onboarding.Enabled && c.Stash().Get("webauthn_available").Bool()) {
-			return c.StartSubFlow(passcode.StatePasscodeConfirmation, passkey_onboarding.StateOnboardingCreatePasskey, shared.StateSuccess)
+			return c.StartSubFlow(passkey_onboarding.StateOnboardingCreatePasskey, shared.StateSuccess)
 		}
 	} else if deps.Cfg.Password.Enabled {
 		return c.ContinueFlow(StatePasswordCreation)
-	} else if !deps.Cfg.Passcode.Enabled {
+	} else if !deps.Cfg.Passcode.Enabled || (deps.Cfg.Passkey.Onboarding.Enabled && c.Stash().Get("webauthn_available").Bool()) {
```

Comment:
Doesn't it have to be like this: 
```suggestion
	} else if !deps.Cfg.Passcode.Enabled && (deps.Cfg.Passkey.Onboarding.Enabled && c.Stash().Get("webauthn_available").Bool()) {
``` 
otherwise the user can still end up in `StateOnboardingCreatePasskey` even when the webauthn API is not available or onboarding is disabled as only passcodes have to be disbaled.